sw/device/lib/dif/dif_padctrl.c

```diff
@@ -0,0 +1,305 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "sw/device/lib/dif/dif_padctrl.h"
+
+// Generated.
+// Note that if the the number of DIO or MIO pads is such that all attribute can
+// fit in one register the PADCTRL_DIO_PADS0_REG_OFFSET and
+// PADCTRL_MIO_PADS0_REG_OFFSET #defines from padctrl_regs.h will no longer
+// contain '0' and become PADCTRL_DIO_PADS_REG_OFFSET and
+// PADCTRL_MIO_PADS_REG_OFFSET respectively which will require altering their
+// names in the source below.
+#include "padctrl_regs.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
+
+_Static_assert(PADCTRL_PARAM_NDIOPADS == 15,
+               "Padctrl instantiation parameters have changed");
+_Static_assert(PADCTRL_PARAM_NMIOPADS == 32,
+               "Padctrl instantiation parameters have changed");
+_Static_assert(PADCTRL_PARAM_ATTRDW == 8,
+               "Padctrl instantiation parameters have changed");
+_Static_assert(PADCTRL_MIO_PADS_ATTR_FIELDS_PER_REG ==
+                   PADCTRL_DIO_PADS_ATTR_FIELDS_PER_REG,
+               "Padctrl instantiation parameters have changed");
+
+static const int kPadctrlRegisterWidthInBytes = (PADCTRL_PARAM_REG_WIDTH / 8);
+
+static const uint32_t kPadctrlAttrFieldMask = PADCTRL_MIO_PADS0_ATTR0_MASK;
```

Comment:
Could you give these constants comments?

Reply:
I felt `kPadctrlRegisterWidthInBytes` was sufficiently clear to not require a comment (especially as that comment would mostly repeat the variable name with some spacing an a couple of extra works), have added one to `kPadctrlAttrFieldMask`